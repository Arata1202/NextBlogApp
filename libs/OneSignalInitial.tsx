'use client';

import { useEffect } from 'react';
import OneSignal from 'react-onesignal';

export const OneSignalInitial = () => {
  useEffect(() => {
    const oneSignalInit = async () => {
      await OneSignal.init({
        appId: process.env.NEXT_PUBLIC_ONESIGNAL_APP_ID || '',
        notifyButton: {
          enable: true,
        },
      }).then(() => {
        OneSignal.Slidedown.promptPush();
      });
    };
    oneSignalInit();
  }, []);
  return (
    <div className="mt-5 flex justify-center">
      <div className="onesignal-customlink-container"></div>
    </div>
  );
};
