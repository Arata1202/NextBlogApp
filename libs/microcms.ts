import { createClient } from 'microcms-js-sdk';
import type {
  MicroCMSQueries,
  MicroCMSImage,
  MicroCMSDate,
  MicroCMSContentId,
} from 'microcms-js-sdk';
import { notFound } from 'next/navigation';

// カテゴリーの型定義
export type Tag = {
  name: string;
} & MicroCMSContentId &
  MicroCMSDate;

// タグの型定義
export type Tag2 = {
  name: string;
} & MicroCMSContentId &
  MicroCMSDate;

// ライターの型定義
export type Writer = {
  name: string;
  profile: string;
  image?: MicroCMSImage;
} & MicroCMSContentId &
  MicroCMSDate;

export type IntroductionBlock = {
  rich_text2?: string;
  custom_html2?: string;
  articleLink3?: string;
  bubble_image?: MicroCMSImage;
  bubble_name?: string;
  bubble_text?: string;
  bubble_isRight?: boolean;
  box_merit?: string;
  box_demerit?: string;
  box_point?: string;
  box_common?: string;
};
export type ContentBlock = {
  adsense?: string;
  rich_text2?: string;
  custom_html2?: string;
  articleLink3?: string;
  bubble_image?: MicroCMSImage;
  bubble_name?: string;
  bubble_text?: string;
  bubble_isRight?: boolean;
  box_merit?: string;
  box_demerit?: string;
  box_point?: string;
  box_common?: string;
};
export type RelatedArticle = {
  articleLink3?: string;
};

// ブログの型定義
export type Blog = {
  title: string;
  description: string;
  thumbnail: MicroCMSImage;
  tags?: Tag[];
  tags2?: Tag2[];
  content_blocks: ContentBlock[];
  introduction_blocks: IntroductionBlock[];
  related_articles?: RelatedArticle[];
};

export type Article = Blog & MicroCMSContentId & MicroCMSDate;

if (!process.env.MICROCMS_SERVICE_DOMAIN) {
  throw new Error('MICROCMS_SERVICE_DOMAIN is required');
}

if (!process.env.MICROCMS_API_KEY) {
  throw new Error('MICROCMS_API_KEY is required');
}

// Initialize Client SDK.
export const client = createClient({
  serviceDomain: process.env.MICROCMS_SERVICE_DOMAIN,
  apiKey: process.env.MICROCMS_API_KEY,
});

// ブログ一覧を取得
export const getList = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getList<Blog>({
      endpoint: 'blog',
      queries,
    })
    .catch(notFound);
  return listData;
};

// ブログの詳細を取得
export const getDetail = async (contentId: string, queries?: MicroCMSQueries) => {
  const detailData = await client
    .getListDetail<Blog>({
      endpoint: 'blog',
      contentId,
      queries,
    })
    .catch(notFound);

  return detailData;
};

// カテゴリーの一覧を取得
export const getCategoryList = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getList<Tag>({
      endpoint: 'tags',
      queries,
    })
    .catch(notFound);

  return listData;
};

// タグの一覧を取得
export const getTagList = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getList<Tag2>({
      endpoint: 'tags2',
      queries,
    })
    .catch(notFound);

  return listData;
};

// カテゴリーの詳細を取得
export const getCategory = async (contentId: string, queries?: MicroCMSQueries) => {
  const detailData = await client
    .getListDetail<Tag>({
      endpoint: 'tags',
      contentId,
      queries,
    })
    .catch(notFound);

  return detailData;
};

// タグの詳細を取得
export const getTag = async (contentId: string, queries?: MicroCMSQueries) => {
  const detailData = await client
    .getListDetail<Tag2>({
      endpoint: 'tags2',
      contentId,
      queries,
    })
    .catch(notFound);

  return detailData;
};
