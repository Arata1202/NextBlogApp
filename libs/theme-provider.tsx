'use client';

import { ThemeProvider as NextThemesProvider } from 'next-themes';
import type { ThemeProviderProps } from 'next-themes';
import { FC, useState, useEffect } from 'react';

export const ThemeProvider: FC<ThemeProviderProps> = (props) => {
  const [isMounted, setIsMounted] = useState(false);

  useEffect(() => {
    setIsMounted(true);
  }, []);

  if (!isMounted) {
    return null;
  }

  return <NextThemesProvider {...props}>{props.children}</NextThemesProvider>;
};
