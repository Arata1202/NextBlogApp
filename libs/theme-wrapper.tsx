'use client';

import { useTheme } from 'next-themes';
import { useEffect } from 'react';

export default function ThemeWrapper() {
  const { theme } = useTheme();

  useEffect(() => {
    document.body.classList.toggle('DarkTheme', theme === 'dark');
  }, [theme]);

  return null;
}
