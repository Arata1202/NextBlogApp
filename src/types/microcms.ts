import type { MicroCMSImage, MicroCMSDate, MicroCMSContentId } from 'microcms-js-sdk';

export type Blog = {
  title: string;
  description: string;
  thumbnail: MicroCMSImage;
  categories: Category[];
  tags?: Tag[];
  introduction_blocks: IntroductionBlock[];
  content_blocks: ContentBlock[];
};

export type Article = Blog & MicroCMSContentId & MicroCMSDate;

export type Category = {
  name: string;
} & MicroCMSContentId &
  MicroCMSDate;

export type Tag = {
  name: string;
} & MicroCMSContentId &
  MicroCMSDate;

export type IntroductionBlock = {
  rich_text?: string;
  custom_html?: string;
  article_link?: string;
  bubble_image?: MicroCMSImage;
  bubble_name?: string;
  bubble_text?: string;
  bubble_isRight?: boolean;
  box_merit?: string;
  box_demerit?: string;
  box_point?: string;
  box_common?: string;
};

export type ContentBlock = {
  google_adsense?: string;
  rich_text?: string;
  custom_html?: string;
  article_link?: string;
  bubble_image?: MicroCMSImage;
  bubble_name?: string;
  bubble_text?: string;
  bubble_isRight?: boolean;
  box_merit?: string;
  box_demerit?: string;
  box_point?: string;
  box_common?: string;
};
