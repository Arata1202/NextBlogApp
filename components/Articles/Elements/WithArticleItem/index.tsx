'use client';
import { useTheme } from 'next-themes';
import React from 'react';
import Image from 'next/image';
import { memo } from 'react';
import { Article } from '@/libs/microcms';
import styles from './index.module.css';
import PublishedDate from '../../../Elements/Date';

type Props = {
  article: Article;
};

function getImageSrcSizes(imageSrc: string) {
  return {
    mobileSrcSet: `${imageSrc}?fm=webp&w=414 1x, ${imageSrc}?fm=webp&w=414&dpr=2 2x`,
    desktopSrcSet: `${imageSrc}?fm=webp&fit=crop&w=240&h=126 1x, ${imageSrc}?fm=webp&fit=crop&w=240&h=126&dpr=2 2x`,
  };
}

const ArticleListItem = ({ article }: Props) => {
  const imageSrc = article.thumbnail?.url || '/no-image.png';
  const isThumbnailAvailable = !!article.thumbnail;
  const { mobileSrcSet, desktopSrcSet } = getImageSrcSizes(imageSrc);

  const handleNavigation = (event: React.MouseEvent<HTMLAnchorElement>, url: string) => {
    event.preventDefault();
    window.location.href = url;
  };

  const { theme } = useTheme();

  const isNextDayOrLater = (date1: string, date2: string) => {
    const d1 = new Date(date1);
    const d2 = new Date(date2);
    d1.setHours(0, 0, 0, 0);
    d2.setHours(0, 0, 0, 0);
    return d1 > d2;
  };

  return (
    <li className={styles.list}>
      <a
        onClick={(event) => handleNavigation(event, `/articles/${article.id}`)}
        href={`/articles/${article.id}`}
        className={`${styles.link} p-2 border shadow-lg hover:shadow-xl transition-shadow duration-200 transform hover:-translate-y-1 cursor-pointer ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}
      >
        <picture>
          {isThumbnailAvailable && (
            <>
              <source type="image/webp" media="(max-width: 640px)" srcSet={mobileSrcSet} />
              <source type="image/webp" srcSet={desktopSrcSet} />
            </>
          )}
          <Image
            src={imageSrc}
            alt="サムネイル"
            className={styles.image}
            width="600"
            height="300"
            placeholder="blur"
            blurDataURL={imageSrc}
          />
        </picture>
        <div className={styles.content}>
          <div className={styles.title}>{article.title}</div>
          <div className={styles.description}>{article.description}</div>
          <div className={styles.date}>
            <PublishedDate date={article.publishedAt!} updatedAt={false} />
            {article.updatedAt && isNextDayOrLater(article.updatedAt, article.publishedAt!) && (
              <>
                &nbsp;&nbsp;&nbsp;&nbsp;
                <PublishedDate date={article.updatedAt!} updatedAt={true} />
              </>
            )}
          </div>
        </div>
      </a>
    </li>
  );
};

export default memo(ArticleListItem);
