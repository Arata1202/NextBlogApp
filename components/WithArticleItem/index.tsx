import React from 'react';
import Image from 'next/image';
import { memo } from 'react';
import { Article } from '@/libs/microcms';
import styles from './index.module.css';
import TagList from '../TagList';
import PublishedDate from '../Date';
import { FolderIcon } from '@heroicons/react/24/outline';

type Props = {
  article: Article;
};

function getImageSrcSizes(imageSrc: string) {
  return {
    mobileSrcSet: `${imageSrc}?fm=webp&w=414 1x, ${imageSrc}?fm=webp&w=414&dpr=2 2x`,
    desktopSrcSet: `${imageSrc}?fm=webp&fit=crop&w=240&h=126 1x, ${imageSrc}?fm=webp&fit=crop&w=240&h=126&dpr=2 2x`,
  };
}

const ArticleListItem = ({ article }: Props) => {
  const imageSrc = article.thumbnail?.url || '/no-image.png';
  const isThumbnailAvailable = !!article.thumbnail;
  const { mobileSrcSet, desktopSrcSet } = getImageSrcSizes(imageSrc);

  const handleNavigation = (url: string) => {
    window.location.href = url;
  };

  return (
    <li className={styles.list}>
      <a
        onClick={() => handleNavigation(`/articles/${article.id}`)}
        href="javascript:void(0)"
        className={`${styles.link} p-2 border border-gray-300 shadow-lg hover:shadow-xl transition-shadow duration-200 transform hover:-translate-y-1 cursor-pointer`}
      >
        <picture>
          {isThumbnailAvailable && (
            <>
              <source type="image/webp" media="(max-width: 640px)" srcSet={mobileSrcSet} />
              <source type="image/webp" srcSet={desktopSrcSet} />
            </>
          )}
          <Image
            src={imageSrc}
            alt="サムネイル"
            className={styles.image}
            width={isThumbnailAvailable ? article.thumbnail?.width : 900}
            height={isThumbnailAvailable ? article.thumbnail?.height : 450}
            // width={isThumbnailAvailable ? article.thumbnail?.width : 1200}
            // height={isThumbnailAvailable ? article.thumbnail?.height : 630}
            placeholder="blur"
            blurDataURL={imageSrc}
          />
        </picture>
        <div className={styles.content}>
          <div className={styles.title}>{article.title}</div>
          <div className={styles.description}>{article.description}</div>
          <div className={styles.date}>
            {/* <FolderIcon className="h-5 w-5 mr-2 mt-3" aria-hidden="true" /> */}
            {/* {article.tags?.map((tag) => (
              <li key={tag.id}>{tag.id}</li>
            ))} */}
            &nbsp;&nbsp;&nbsp;&nbsp;
            <PublishedDate date={article.publishedAt || article.createdAt} />
          </div>
        </div>
      </a>
    </li>
  );
};

export default memo(ArticleListItem);
