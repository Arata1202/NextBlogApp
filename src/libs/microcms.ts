import { notFound } from 'next/navigation';
import { createClient } from 'microcms-js-sdk';
import type { MicroCMSQueries } from 'microcms-js-sdk';
import { Blog, Category, Tag } from '@/types/microcms';

if (!process.env.MICROCMS_SERVICE_DOMAIN) {
  throw new Error('MICROCMS_SERVICE_DOMAIN is required');
}

if (!process.env.MICROCMS_API_KEY) {
  throw new Error('MICROCMS_API_KEY is required');
}

export const client = createClient({
  serviceDomain: process.env.MICROCMS_SERVICE_DOMAIN,
  apiKey: process.env.MICROCMS_API_KEY,
});

export const getAllLists = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getAllContents<Blog>({
      endpoint: 'blog',
      queries,
    })
    .catch(notFound);

  return listData;
};

export const getList = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getList<Blog>({
      endpoint: 'blog',
      queries,
    })
    .catch(notFound);

  return listData;
};

export const getDetail = async (contentId: string, queries?: MicroCMSQueries) => {
  const detailData = await client
    .getListDetail<Blog>({
      endpoint: 'blog',
      contentId,
      queries,
    })
    .catch(notFound);

  return detailData;
};

export const getAllCategoryLists = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getAllContents<Category>({
      endpoint: 'categories',
      queries,
    })
    .catch(notFound);

  return listData;
};

export const getCategoryList = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getList<Category>({
      endpoint: 'categories',
      queries,
    })
    .catch(notFound);

  return listData;
};

export const getCategory = async (contentId: string, queries?: MicroCMSQueries) => {
  const detailData = await client
    .getListDetail<Category>({
      endpoint: 'categories',
      contentId,
      queries,
    })
    .catch(notFound);

  return detailData;
};

export const getAllTagLists = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getAllContents<Tag>({
      endpoint: 'tags',
      queries,
    })
    .catch(notFound);

  return listData;
};

export const getTagList = async (queries?: MicroCMSQueries) => {
  const listData = await client
    .getList<Tag>({
      endpoint: 'tags',
      queries,
    })
    .catch(notFound);

  return listData;
};

export const getTag = async (contentId: string, queries?: MicroCMSQueries) => {
  const detailData = await client
    .getListDetail<Tag>({
      endpoint: 'tags',
      contentId,
      queries,
    })
    .catch(notFound);

  return detailData;
};
