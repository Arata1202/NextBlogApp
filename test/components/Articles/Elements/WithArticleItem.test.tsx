import { render } from '@testing-library/react';
import WithArticleItem from '@/components/Articles/Elements/WithArticleItem';
import { Article } from '@/libs/microcms';

const article: Article = {
  id: '1',
  title: 'タイトル',
  description: '概要',
  thumbnail: { url: 'https://thumbnail.com/thumbnail.jpg' },
  publishedAt: '2023-10-30T12:00:00Z',
  content_blocks: [],
  introduction_blocks: [],
  createdAt: '2023-10-30T12:00:00Z',
  updatedAt: '2023-10-30T12:00:00Z',
};

describe('WithArticleItem', () => {
  test('スナップショット（WithArticleItem）', () => {
    const { asFragment } = render(<WithArticleItem article={article} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
