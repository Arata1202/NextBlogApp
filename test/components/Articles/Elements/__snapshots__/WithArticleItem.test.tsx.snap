// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithArticleItem スナップショット（WithArticleItem） 1`] = `
<DocumentFragment>
  <li
    class="list"
  >
    <a
      class="link p-2 border border-gray-300 shadow-lg hover:shadow-xl transition-shadow duration-200 transform hover:-translate-y-1 cursor-pointer"
      href="/articles/1"
    >
      <picture>
        <source
          media="(max-width: 640px)"
          srcset="https://thumbnail.com/thumbnail.jpg?fm=webp&w=414 1x, https://thumbnail.com/thumbnail.jpg?fm=webp&w=414&dpr=2 2x"
          type="image/webp"
        />
        <source
          srcset="https://thumbnail.com/thumbnail.jpg?fm=webp&fit=crop&w=240&h=126 1x, https://thumbnail.com/thumbnail.jpg?fm=webp&fit=crop&w=240&h=126&dpr=2 2x"
          type="image/webp"
        />
        <img
          alt="サムネイル"
          class="image"
          data-nimg="1"
          decoding="async"
          height="300"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fthumbnail.com%2Fthumbnail.jpg&w=1200&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fthumbnail.com%2Fthumbnail.jpg&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fthumbnail.com%2Fthumbnail.jpg&w=1200&q=75 2x"
          style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
          width="600"
        />
      </picture>
      <div
        class="content"
      >
        <div
          class="title"
        >
          タイトル
        </div>
        <div
          class="description"
        >
          概要
        </div>
        <div
          class="date"
        >
              
        </div>
      </div>
    </a>
  </li>
</DocumentFragment>
`;
